### PR TITLE
Strip unneeded for shared libraries

### PR DESCRIPTION
**Pull request by chunseoklee**
- Strip with --strip-unneeded option for libraries

ONE-DCO-1.0-Signed-off-by: Chunseok Lee <chunseok.lee@samsung.com>

### CODE REVIEW

**Comment by hseok-oh** [comment]
@chunseoklee This PR disables debugging on debug build. Could you change this PR to not strip on debug build?

**Review by glistening — APPROVED** [review]
LGTM

**Review by YongseopKim — APPROVED** [review]
LGTM

File: runtime/onert/api/CMakeLists.txt [hunk 1]
```diff
@@ -22,6 +22,11 @@ endif (ANDROID)
 target_include_directories(${ONERT_DEV} PUBLIC include)
 set_target_properties(${ONERT_DEV} PROPERTIES PUBLIC_HEADER "${NNFW_API_HEADERS}")
 
+if(CMAKE_BUILD_TYPE STREQUAL "release")
```
Comment (hseok-oh) [inline runtime/onert/api/CMakeLists.txt:25]: It needs to handle upper/lower case. (ex. "Release", "RELEASE", etc)

**Review by hseok-oh — APPROVED** [review]

File: runtime/onert/api/CMakeLists.txt (hunk 1 above)
Comment (chunseoklee) [reply runtime/onert/api/CMakeLists.txt:25]: Our build process takes type strings other than "release" ?

**Pull request merged**